app/main.py

```diff
@@ -1,8 +1,19 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(peopl: list) -> list[Person]:
+    person_list = [Person(person["name"], person["age"]) for person in peopl]
+    for person in peopl:
+        if person.get("husband"):
```

Comment:
also recommend use https://peps.python.org/pep-0572/#exceptional-cases

```suggestion
        if husband := person.get("husband"):
             Person.people[person.get("name")].husband= Person.people[husband]   
```